### PR TITLE
Change delay logic for buses delayed for < 120 seconds

### PR DESCRIPTION
**Pull request by ywsang**
Addresses #212.

### CODE REVIEW

**Review by Omarrasheed — APPROVED** [review]
lgtm

File: TCAT/Cells/RouteTableViewCell.swift [hunk 1]
```diff
@@ -264,10 +264,14 @@ class RouteTableViewCell: UITableViewCell {
                 
                 let delayedDepartTime = departTime.addingTimeInterval(TimeInterval(delay))
                 
-                if Time.compare(date1: departTime, date2: delayedDepartTime) == .orderedAscending {
-                    return .late(date: delayedDepartTime)
+                if Time.compare(date1: departTime, date2: delayedDepartTime) == .orderedAscending { // bus is delayed
+                    if (delayedDepartTime >= Date() || delay >= 120) {
+                        return .late(date: delayedDepartTime)
+                    } else { // delay < 120
+                        return .onTime(date: departTime) // what time??
```
Comment (kevinchan159) [inline TCAT/Cells/RouteTableViewCell.swift]: what does the `// what time?` comment mean

**Review by kevinchan159 — APPROVED** [review]

File: TCAT/Cells/RouteTableViewCell.swift (hunk 1 above)
Comment (ywsang) [reply TCAT/Cells/RouteTableViewCell.swift]: whoops it was for when I was asking you what time to display in that case, I just removed it

**Review by mattbarker016 — COMMENTED** [review]
Just to confirm, we're technically going to show buses as time even when they're up 2 min late?

I'm sorry to beat a dead horse a bit, but can you make an extra explicit comment in here that explains what this means? "If buses are a bit late, they possibly won't show in the route list as a valid bus to take, even though it might catch up." This check makes sure buses are definitively late.

Is there a reason we considered 120 over 60 sec also?

**Comment by kevinchan159** [comment]
> Just to confirm, we're technically going to show buses as time even when they're up 2 min late?
> 
> I'm sorry to beat a dead horse a bit, but can you make an extra explicit comment in here that explains what this means? "If buses are a bit late, they possibly won't show in the route list as a valid bus to take, even though it might catch up." This check makes sure buses are definitively late.
> 
> Is there a reason we considered 120 over 60 sec also?

@mattbarker016 The reason is because our live tracking only updates once every 30 seconds which means that within a 60 second timespan, it could be that the location only gets updated once. By having a greater threshold, we'll be more accurate about this. Plus, this means that riders will always get to the stop earlier than later which is better than missing the bus

**Comment by ywsang** [comment]
@mattbarker016 An explicit comment in the code itself?

**Comment by mattbarker016** [comment]
@ywsang I would include the first part of Kevin's above comment in the code. @kevinchan159 So why not making the constant 60 sec?

**Pull request merged**